src/util/user-dirs.js

```diff
@@ -0,0 +1,45 @@
+/* @flow */
+
+const path = require('path');
+const userHome = require('./user-home-dir').default;
+
+export function getDataDir(): string {
+  if (process.platform === 'win32') {
+    return path.join(getLocalAppDataDir(), 'Data');
+  } else if (process.env.XDG_DATA_HOME) {
+    return path.join(process.env.XDG_DATA_HOME, 'yarn');
+  } else {
+    // This could arguably be ~/Library/Application Support/Yarn on Macs,
+    // but that feels unintuitive for a cli tool
+
+    // Instead, always use the typical location of XDG_DATA_HOME
+    return path.join(userHome, '.local', 'share', 'yarn');
```

Comment:
I have a concern that this changes current setup people have, may cause confusion.

Before it was `~/.config/yarn/global`, now it will be `~/.local/share/yarn/global`.
Do we really need this change?

Reply:
Honestly this is my main reason for the PR 😝 

I thought it was really, really weird to be storing a ton of binaries in the user's *configuration* directory. The kind of stuff we put here is a perfect fit with XDG_DATA_HOME, which on many machines defaults to this location.